test/gtest/ucp/ucp_test.cc

```diff
@@ -349,8 +349,13 @@ void ucp_test::add_variant_values(std::vector<ucp_test_variant>& variants,
 
 void ucp_test::add_variant_memtypes(std::vector<ucp_test_variant>& variants,
                                     get_variants_func_t generator,
-                                    uint64_t mem_types_mask)
+                                    uint64_t mem_types_mask,
+                                    bool memtype_support_check)
 {
+    if (memtype_support_check && !mem_buffer::is_gpu_supported()) {
+        return;
+    }
+
```

Comment:
if both gpu and host are supported, it still takes host?
should be
bool gpu_only
mem_types = mem_buffer::supported_mem_types(gpu_only)

Reply:
i want to create (and test) combinations, like `host<->gpu`